### ISSUE TITLE
perform multiple calls in 1 request

### ISSUE THREAD

**Issue by cwinters8** [cwinters8/gomap#3]
It would be nice to be able to provide a way to chain multiple method calls together into a single request when possible.

When making requests to Fastmail's API directly, you can do something like

```json
[[ "Email/query",{
  "accountId": "ue150411c",
  "filter": {
    "inMailbox": "fb666a55"
  },
  "sort": [{
    "isAscending": false,
    "property": "receivedAt"
  }],
  "collapseThreads": true,
  "position": 0,
  "limit": 30,
  "calculateTotal": true
}, "0" ],
[ "Email/get", {
  "accountId": "ue150411c",
  "#ids": {
    "resultOf": "0",
    "name": "Email/query",
    "path": "/ids"
  },
  "properties": [
    "threadId"
  ]
}, "1" ]]
```
([source](https://jmap.io/spec-mail.html#examples))

to both reduce the number of required HTTP requests and to be able to pass values between method calls.

**Comment by cwinters8** [comment]
This should be possible now that a [requests.Request takes a slice of Call interface values](https://github.com/cwinters8/gomap/blob/main/requests/request.go#L16), but still needs to be validated.

**Comment by cwinters8** [comment]
Still working out how to refer to the output of a prior method in the slice of Calls. One idea is to pass a slice of some kind of `Reference` struct to a Call that would define what field referred to a prior call.

**Comment by cwinters8** [comment]
A first step toward implementing a reference architecture as described in the prior comment has been made with commit d1ada3cf079c2527a41c34f0b50883ffc346a525 - this implements the ability to retrieve multiple emails in a single call + request, which will be needed to validate references. Next step will be to implement the `Reference` struct, then attempt chaining multiple calls of different types and referring to values from an earlier call.

**Comment by cwinters8** [comment]
It turns out the ability to utilize references between a given request's calls is very limited. I tried many approaches, and the only thing that seems to work is in the very specific case when you query for a bunch of IDs and then you want to get data about all of them. You cannot, for example, make a `set` request and then try to `get` the created object because the ID that is returned by the set request is not usable in the get request, as `get` requires a list of IDs, not 1 single ID.

Conclusion: Implementing anything specific for referring to data between calls is not worth it. It can still be done by modifying the body for a Call to [use a reference as described in the spec](https://jmap.io/spec-core.html#references-to-previous-method-results), but users should be warned it is highly error prone, and that it is usually better to just make another request to get the data you need.

I am working on a test that validates multiple calls can be made in a request, but I will not be pursuing the reference architecture.

**Comment by cwinters8** [comment]
Multiple requests test implemented with commit 27928d54ee393f04c8576ea58e1f1fac3e089735
